### PR TITLE
8301132: Test update for deprecated sprintf in Xcode 14

### PR DESCRIPTION
**Pull request by XueleiFan**
The sprintf is deprecated in Xcode 14 because of security concerns. The issue was addressed in [JDK-8296812](https://bugs.openjdk.org/browse/JDK-8296812)/[JDK-8299378](https://bugs.openjdk.org/browse/JDK-8299378)/[JDK-8299635](https://bugs.openjdk.org/browse/JDK-8299635), but the test was not fully covered yet. This would be the last effort to fix the issue in test.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301132](https://bugs.openjdk.org/browse/JDK-8301132): Test update for deprecated sprintf in Xcode 14

### Reviewers
 * [Mikael Vidstedt](https://openjdk.org/census#mikael) (@vidmik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12211/head:pull/12211` \
`$ git checkout pull/12211`

Update a local copy of the PR: \
`$ git checkout pull/12211` \
`$ git pull https://git.openjdk.org/jdk pull/12211/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12211`

View PR using the GUI difftool: \
`$ git pr show -t 12211`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12211.diff">https://git.openjdk.org/jdk/pull/12211.diff</a>

</details>

### CODE REVIEW

File: test/jdk/sun/management/windows/exerevokeall.c [hunk 1]
```diff
@@ -99,12 +99,13 @@ static char *getTextualSid(SID* sid) {
     }
 
     // S-SID_REVISION
-    sprintf(name, "S-%lu-", SID_REVISION );
+    snprintf(name, sizeof(name), "S-%lu-", SID_REVISION );
```
Comment (vidmik) [inline test/jdk/sun/management/windows/exerevokeall.c:102]: Name is a `char *` so sizeof will return the size of the pointer, not the length of the allocate data.

Comment (XueleiFan) [reply test/jdk/sun/management/windows/exerevokeall.c]: Ooooops.  Thank you for the catch!

**Review by vidmik — APPROVED** [review]

**Pull request closed without merging**